crates/sui-types/src/object.rs

```diff
@@ -525,11 +529,26 @@ impl Object {
         previous_transaction: TransactionDigest,
         max_move_package_size: u64,
     ) -> Result<Self, ExecutionError> {
+        let type_origin = BTreeMap::from_iter(modules.iter().flat_map(|module| {
+            module.struct_handles().iter().map(|struct_handle| {
+                let module_name = Identifier::from(module.name());
+                let struct_name = Identifier::from(module.identifier_at(struct_handle.name));
+                (
+                    ModuleStruct {
+                        module_name,
+                        struct_name,
+                    },
+                    ObjectID::from(SuiAddress::from(*module.self_id().address())),
```

Comment:
You may just be able to do `module.address().into()` -- also we can just do this once at the top.